.dir-locals.el

```diff
@@ -0,0 +1,2 @@
+((haskell-mode
+  (haskell-stylish-on-save . t)))
```

Comment:
It's fairly common, e.g.  https://github.com/ghc/ghc/blob/master/rts/.dir-locals.el

It's actually not a program, it's a list of variable assignments.  It's only considered safe it if sets variables that are marked safe.

That being said, it's usually used for things like indentation style.  This is on the edge of something I'd consider a personal setting...